pkg/gpu/nvidia/alpha_plugin.go

```diff
@@ -49,9 +51,23 @@ func (s *pluginServiceV1Alpha) ListAndWatch(emtpy *pluginapi.Empty, stream plugi
 }
 
 func (s *pluginServiceV1Alpha) Allocate(ctx context.Context, rqt *pluginapi.AllocateRequest) (*pluginapi.AllocateResponse, error) {
+	// Validate if it is requesting time-sharing GPU resources.
+	// If it is, then validate if the request meets the time-sharing specific conditions.
+	if err := timesharing.ValidateRequest(rqt.DevicesIDs, len(s.ngm.ListPhysicalDevices())); err != nil {
+		return nil, err
+	}
 	resp := new(pluginapi.AllocateResponse)
 	// Add all requested devices to Allocate Response
 	for _, id := range rqt.DevicesIDs {
+		// If we are using the time-sharing solution, the input deviceID will be a virtual Device ID.
+		// We need to map it to the corresponding physical device ID.
+		if timesharing.IsEnabled(s.ngm.gpuConfig.GPUSharingConfig.GPUSharingStrategy) {
+			physicalDeviceID, err := timesharing.VirtualToPhysicalDeviceID(id)
+			if err != nil {
+				return nil, err
+			}
+			id = physicalDeviceID
+		}
```

Comment:
I notice that this logic is already beginning to diverge from the beta implementation. We should probably deprecate this soon so that it does not add more complexity. No action needed for this PR though.

Reply:
Yeah, I agree with that. I'll open a bug for that.